### ISSUE TITLE
Nested Struct Validation Fails

### ISSUE THREAD

**Issue by safaci2000** [go-playground/validator#1044]
- [ ] I have looked at the documentation [here](https://pkg.go.dev/github.com/go-playground/validator/v10#section-documentation) first? Yes
- [ ] I have looked at the examples provided that may showcase my question [here](/_examples)?
Yes
### Package version eg. v9, v10: 
v10

### Issue, Question or Enhancement:
I have a nested struct as demonstrated below and when I utilize this type of structure validation fails. 

### Code sample, to showcase or reproduce:

```go
package common

type Equipment struct {
	ID            int32           `json:"id"`
	URL           string          `json:"url"`
	Name          string          `json:"name"`
	Hostname      string          `json:"hostname"`
	Network       Network         `json:"network"`
	State         State           `json:"equipment_state"`
	OS            OS              `json:"os"`
	Role          Role            `json:"role"`
	Platform      Platform        `json:"platform"`
}

package model

//Equipment Represent the data model returned from ESDB
type Equipment struct {
	common.Equipment
	OutputModel OutputType
}

```

Basically the struct in model extends the object defined in common and I use it in my code.

If I create a struct using the common entity it works fine but if I use the model version it fails.  Here's the code I used to test the validation.

```go
	obj := common.Equipment{}
	obj.Role = 5

	v := validator.New()
	rules := map[string]string{
		"Role": "min=99,max=200",
	}
	v.RegisterStructValidationMapRules(rules, common.Equipment{})

	err := v.Struct(obj)
	if err != nil {
		for _, e := range err.(validator.ValidationErrors) {
			fmt.Println(e)
		}
	}

```
